Sources/FrontEnd/TypeChecking/TypeChecker+ConstraintGeneration.swift

```diff
@@ -8,33 +8,39 @@ extension TypeChecker {
   struct InferenceFacts {
```

Comment:
```suggestion
  struct Inferences {
```

or maybe:

```suggestion
  struct InferenceSet {
```

IIUC, these "facts" are not necessarily true facts as we'll explore inferences that turn out not to work out in the end.  Aren't they just inferences?

Reply:
I picked this name from the world of logic programming. In Prolog, for example, a program is often described as a based of facts (and rules). In a sense, the generator is building small logic programs that are "ran" by the the constraint solver. Rules are hard-coded in our cases; they correspond to the rules of Val's type system. So we're only producing facts.

That being said, I'll concede the final say to the English speaker.

Reply:
Except that, IIUC, prolog will immediately reject any facts that cause conflicts so that the experience of the user is that all these things actually are facts, while we speculatively explore "facts" that cause conflicts.   I think that term is an appropriate description for the Prolog user.